### PR TITLE
Bnb fixes

### PR DESCRIPTION
**Pull request by Wingman4l7**
##### Checklist

- [x] `./tools.sh full-test` passes
- [x] tests are included

### CODE REVIEW

File: dragonchain/lib/dto/bnb.py [hunk 1]
```diff
@@ -275,36 +275,47 @@ def _fetch_account(self):
...
+            symbol (str): the exchange symbol for the token
+            memo (str): string of data to publish in the transaction
+        Returns:
+            dict of the constructed transaction
+        """
+        amount = raw_transaction["amount"]
+        if amount <= 0:
+            raise exceptions.BadRequest("[BINANCE] Amount in transaction cannot be less than or equal to 0.")
 
+        symbol = raw_transaction["symbol"]
```
Comment (cheeseandcereal) [inline dragonchain/lib/dto/bnb.py]: you assume symbol exists, but it's not required in your schema

Comment (Wingman4l7) [reply dragonchain/lib/dto/bnb.py]: It's not required, but it is defaulted to "BNB" if it is not provided -- see `dragonchain_client.py`.

Comment (cheeseandcereal) [reply dragonchain/lib/dto/bnb.py]: yeah, but the sdk isn't how you should rely on a default. What if someone else writes a dragonchain client? Now the Dragonchain can return a 500 error if you don't know that you have to provide a symbol, which isn't required by the schema on Dragonchain.

You would expect not providing some part of the schema to result in a 400, not an unhandled 500.

Defaulting a value in the SDK is a nice-to-have, but is not a substitute for making sure the back-end accepts and handles good/bad/missing values.

An api is a user interface too. You definitely shouldn't make the assumption that everyone will interact with the Dragonchain via only our sanctioned SDKs, and the backend should be able to handle bad input.

So we need to either require symbol in the schema, or default it on the back-end if it's not provided.

Comment (Wingman4l7) [reply dragonchain/lib/dto/bnb.py]: Made sure that `_build_transaction_msg()` function in the backend provided the proper defaults.  Removed defaults from SDK, feels redundant there.

File: dragonchain/lib/dto/bnb.py [hunk 2]
```diff
@@ -275,36 +275,46 @@ def _fetch_account(self):
...
+            symbol (str, optional): the exchange symbol for the token (defaults to 'BNB')
+            memo (str, optional): string of data to publish in the transaction (defaults to '')
+        Returns:
+            dict of the constructed transaction
+        """
+        amount = raw_transaction["amount"]
+        if amount <= 0:
+            raise exceptions.BadRequest("[BINANCE] Amount in transaction cannot be less than or equal to 0.")
+
+        symbol = raw_transaction["symbol"] if raw_transaction.get("symbol") is not None else "BNB"
```
Comment (cheeseandcereal) [inline dragonchain/lib/dto/bnb.py]: shorthand:
```suggestion
        symbol = raw_transaction.get("symbol") or "BNB"
```

File: dragonchain/lib/dto/bnb.py [hunk 3]
```diff
@@ -275,36 +275,46 @@ def _fetch_account(self):
...
+        memo = raw_transaction["memo"] if raw_transaction.get("memo") is not None else ""
```
Comment (cheeseandcereal) [inline dragonchain/lib/dto/bnb.py]: shorthand:
```suggestion
        memo = raw_transaction.get("memo") or ""
```

File: dragonchain/lib/dto/bnb.py (hunk 2 above)
Comment (Wingman4l7) [reply dragonchain/lib/dto/bnb.py]: Fixed!

File: dragonchain/lib/dto/bnb.py (hunk 3 above)
Comment (Wingman4l7) [reply dragonchain/lib/dto/bnb.py]: Fixed!

**Pull request merged**